File: Dev/C++/Mascaret/src/VEHA/Kernel/LiteralReal.cpp
#include "VEHA/Kernel/LiteralReal.h"
#include "VEHA/Kernel/PrimitiveType.h"
#include "MascaretApplication.h"
#include "VEHA/Kernel/Model.h"

#include <sstream>

namespace VEHA
{

LiteralReal::LiteralReal(const double& value)
:LiteralSpecification(MascaretApplication::getInstance()->getModel()->getBasicType("real")),_value(value)
{
}
LiteralReal::LiteralReal(const string & str)
:LiteralSpecification(MascaretApplication::getInstance()->getModel()->getBasicType("real")),_value(boost::lexical_cast<double>(str))
{
}

LiteralReal::~LiteralReal()
{
}
bool LiteralReal::getDoubleValue(double & value) {value = _value; return true;}
//bool LiteralReal::setDoubleValue(const double & value) {_value = value; return true;}

bool LiteralReal::setValueFromString(const string& value)
{
	try
	{
		_value= boost::lexical_cast<double>(value);
		return true;
	}
	catch(boost::bad_lexical_cast &)
	{
		return false;
	}
	return true;
}

string LiteralReal::getStringFromValue() const
{
	return   boost::lexical_cast<std::string>(_value);
}
LiteralReal::operator double () const
        		{
	return _value;
        		}
LiteralReal::operator string () const
        		{
	return  boost::lexical_cast<std::string>(_value);
        		}

//const ValueSpecification& LiteralReal::operator= (double value)
//{
//	cerr<<"= operator LiteralReal"<<endl;
//	_value=value;
//	cerr<<"_value="<<_value<<endl;
//	return *this;
//}
shared_ptr<ValueSpecification> LiteralReal::clone() const
{
	LiteralReal * vs=new LiteralReal();
	vs->CLASSTYPE=CLASSTYPE;
	vs->_type=_type;
	vs->_value=_value;
	return shared_ptr<ValueSpecification>(vs);
}
//const ValueSpecification& LiteralReal::operator= (const ValueSpecification&vs)
//{
//	_value=(double)vs;
//	return *this;
//}

}

File: Dev/CS/Mascaret/Mascaret/VEHA/Kernel/LiteralReal.cs
using System;
using System.Collections.Generic;


namespace Mascaret
{
    public class LiteralReal : LiteralSpecification
    {


        protected double rValue;
        public double RValue
        {
            get { return rValue; }
            set { rValue = value; }
        }

        public bool setValueFromString(string value)
        {
            rValue = double.Parse(value);
            return true;
        }

        public override string getStringFromValue()
        {
            return (rValue.ToString());
        }

        public override double getDoubleFromValue()
        {
            return rValue;
        }

        public LiteralReal()
            : base(MascaretApplication.Instance.Model.getBasicType("real"))
        {
        }

        public LiteralReal(string svalue)
            : base(MascaretApplication.Instance.Model.getBasicType("real"))
        {
            setValueFromString(svalue);
        }

        public LiteralReal(double r)
            : base(MascaretApplication.Instance.Model.getBasicType("real"))
        {
            rValue = r;
        }

        public override ValueSpecification clone()
        {
            LiteralReal vs = new LiteralReal(0.0);
            vs.CLASSTYPE = CLASSTYPE;
            vs.Type = Type;
            vs.rValue = rValue;
            return ((ValueSpecification)vs);
        }

    }
}

